tests/unit/test_index.py

```diff
@@ -1,437 +1,464 @@
 import pandas as pd
-import numpy as np
 import pytest
 import warnings
 
-from pinecone.core.client.api_client import Endpoint
-
 import pinecone
+from pinecone import UpsertRequest, Vector
 from pinecone import DescribeIndexStatsRequest, ScoredVector, QueryResponse, UpsertResponse, SparseValues
 
 
 class TestRestIndex:
-
     def setup_method(self):
         self.vector_dim = 8
-        self.id1 = 'vec1'
-        self.id2 = 'vec2'
+        self.id1 = "vec1"
+        self.id2 = "vec2"
         self.vals1 = [0.1] * self.vector_dim
         self.vals2 = [0.2] * self.vector_dim
-        self.md1 = {'genre': 'action', 'year': 2021}
-        self.md2 = {'genre': 'documentary', 'year': 2020}
-        self.filter1 = {'genre': {'$in': ['action']}}
-        self.filter2 = {'year': {'$eq': 2020}}
+        self.md1 = {"genre": "action", "year": 2021}
+        self.md2 = {"genre": "documentary", "year": 2020}
+        self.filter1 = {"genre": {"$in": ["action"]}}
+        self.filter2 = {"year": {"$eq": 2020}}
         self.svi1 = [1, 3, 5]
         self.svv1 = [0.1, 0.2, 0.3]
         self.sv1 = {
-            'indices': self.svi1,
-            'values': self.svv1,
+            "indices": self.svi1,
+            "values": self.svv1,
         }
         self.svi2 = [2, 4, 6]
         self.svv2 = [0.1, 0.2, 0.3]
-        self.sv2 = {
-            'indices': self.svi2,
-            'values': self.svv2
-        }
+        self.sv2 = {"indices": self.svi2, "values": self.svv2}
 
-        pinecone.init(api_key='example-key')
-        self.index = pinecone.Index('example-name')
-    
-    def test_upsert_numpy_deprecation_warning(self, mocker):
```

Comment:
**question:** Are we still planning on deprecating numpy? I've seen references to that in the codebase and this seems to confirm that. 

If so we can probably add a ticket.

Reply:
Yeah, we should remove it before releasing spruce. I'll add a ticket.